src/games/strategy/triplea/ui/AbstractUIContext.java

```diff
@@ -297,33 +297,24 @@ public void shutDown() {
     return rVal;
   }
 
-  private static Map<String,String> getSkins(final String mapName) {
+  private static Map<String, String> getSkins(final String mapName) {
     final Map<String, String> rVal = new HashMap<>();
     final File[] files = ClientFileSystemHelper.getUserMapsFolder().listFiles();
     if (files == null) {
       return rVal;
     }
-
     for (final File f : files) {
-      if (!f.isDirectory()) {
-        if (mapSkinNameMatchesMapName(f.getName(), mapName)) {
-          final String skinName = f.getName().substring(0, f.getName().indexOf('-'));
-          if (f.getName().endsWith(".zip")) {
-            rVal.put(skinName, f.getName());
-          } else {
-            rVal.put(f.getName().substring(f.getName().indexOf('-') + 1), f.getName());
-          }
-        }
+      if (mapSkinNameMatchesMapName(f.getName(), mapName)) {
+        final String displayName = f.getName().replace(mapName + "-", "").replace("-master", "").replace(".zip", "");
```

Comment:
It removes "mapName-" from the beginning then "-master" and ".zip" from the end. I don't believe there are any issues unless for some reason "-master" would appear in the map name (we are treating that as something that should really be used since we add it to the end of all map downloads).

You're welcome to add tests. This code is terrible and there should be a flag or enum to indicate whether its a skin not parsing the name.